File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.py
def solution(s):
    r,al=-1,[]
    # 스택을 이용해 구현
    for c in s:
        if not len(al):
            al+=[c]
        else:
            if al[-1]==c:
                del al[-1]
            else:
                al+=[c]
    r=1 if not len(al) else 0
    return r

File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.cpp
#include <vector>
#include <string>

using namespace std;

int solution(string s)
{
    int r=-1;
    vector<char> al;
    for(int i=0;i<s.length();i++) {
        if(al.size()==0)
            al.push_back(s[i]);
        else {
            if(al.back()==s[i])
                al.erase(al.begin()+al.size()-1);
            else
                al.push_back(s[i]);
        }
    }
    return r=(al.size()==0)?1:0;
}